src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -129,10 +129,34 @@ internal void Remove(T e)
             }
         }
 
+        // Enusre full cache line occupied so no false sharing with other work items
+        [StructLayout(LayoutKind.Explicit, Size = 64)]
+        internal struct PaddedWorkItem
+        {
+            [FieldOffset(0)]
+            public IThreadPoolWorkItem Item;
+        }
+
+        // Enusre full cache line occupied so no false sharing with other work items
+        [StructLayout(LayoutKind.Explicit, Size = 64)]
+        internal struct PaddedQueueSegment
+        {
+            [FieldOffset(0)]
+            public volatile QueueSegment Segment;
+        }
+
+        // Enusre full cache line occupied so no false sharing with other indicies
+        [StructLayout(LayoutKind.Explicit, Size = 64)]
+        internal struct PaddedInt
+        {
+            [FieldOffset(0)]
+            public volatile int Value;
+        }
+
         internal class WorkStealingQueue
         {
             private const int INITIAL_SIZE = 32;
-            internal volatile IThreadPoolWorkItem[] m_array = new IThreadPoolWorkItem[INITIAL_SIZE];
+            internal volatile PaddedWorkItem[] m_array = new PaddedWorkItem[INITIAL_SIZE];
```

Comment:
Why are you 64-byte padding the items in the queue?  The queue is owned by a single thread, and all other threads need to take a lock to access it.  The owning thread also needs to take a lock when there are fewer than two items in the queue.  The only case where you'd have contention this would help with is if there was a thread stealing concurrently with the owning thread pushing/popping on a list with at least two elements.  At that point, they're already some distance apart, though not necessarily a full cache line.  Have you shown that this change makes a notable improvement?  It does so at the expense of effectively increasing the size of every work item by 56 bytes on 64-bit, since every work item reference to be stored now consumes 64 bytes instead of 8 bytes (plus the size of the work item object itself).